test/parallel_api/ranges/transform_ranges_factory_sycl.pass.cpp

```diff
@@ -22,20 +22,23 @@
 #endif
 
 #include "support/utils.h"
+#include "support/utils_invoke.h" // for CLONE_TEST_POLICY macro
 
+#include <algorithm> // std::transform
 #include <iostream>
 
-std::int32_t
-main()
-{
 #if _ENABLE_RANGES_TESTING
+template <typename Policy>
+void
+test_impl(Policy&& exec)
+{
     constexpr int max_n = 10;
     int data[max_n] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
     int data2[max_n];
     int data3[max_n];
 
-    auto lambda1 = [](auto i) { return i * i; };
-    auto lambda2 = [](auto i) { return i + 200; };
+    auto lambda1 = TestUtils::Pow2<int>();
+    auto lambda2 = TestUtils::SumWithOp<int>{200};
```

Comment:
A yet another nitpick optional to address: these are not lambdas anymore, so perhaps, it is better to rename them.